### PR TITLE
Advanced manual derivation of converters

### PR DESCRIPTION
**Pull request by jendakol**
The motivation is to improve and describe custom derivation of converters to support advanced user scenarios.

### CODE REVIEW

**Comment by jendakol** [comment]
Well, I hoped you will try to understand it from the example, because if you don't, probably nobody will :smile:  
I believe it is real use case, it came out from one situation (not for the first time) and it was generalized a little bit.

**Comment by augi** [comment]
I tried to understand it from the example and I cannot say that I´m 100% sure what´s happening there :(

This library is basically about converting Java class to Scala case-class. I don't think that this problem is inherently so hard and so I'm not sure if this library has to be so complicated (even to use).

**Maybe it would help to have shorter README for stupid people like me, with the very basic usage shown where implementation details wouldn't leak to the library user?**
If I were a developer from outside, I would be scared by the current README and rather convert Java classes to Scala case-classes manually (it's actually what I'm doing in one of _my_ projects).

**Comment by jakubjanecek** [comment]
So as discussed personally I suggest using plain words to explain the potential use case and also simplify the examples to the bare minimum (e.g. remove GPB Any). That should make it more clear.

**Comment by jendakol** [comment]
I updated the _advanced scenario_ to something much more easy but still worth to mention. Check it, please.

File: README.md [hunk 1]
```diff
@@ -368,8 +368,75 @@ the application
...
+}
+```
+
+```scala
+case class OptionalUserResponse(user: Option[User])
+
+case class User(name: String, age: Int)
+```
+
+Now, the Cactus is able to convert `OptionalUserResponse` to `OptionalUserResponseMessage` (and back), but you want to have just `Option[User]`
```
Comment (petr-k) [inline README.md]: just Cactus, no _the_

**Pull request merged**